qt/python/mantidqt/widgets/instrumentview/presenter.py

```diff
@@ -13,15 +13,52 @@
 from __future__ import (absolute_import, unicode_literals)
 
 # local imports
+from mantidqt.widgets.common.observing_presenter import ObservingPresenter
+from mantidqt.widgets.common.workspacedisplay_ads_observer import WorkspaceDisplayADSObserver
 from .view import InstrumentView
 
 
-class InstrumentViewPresenter(object):
+class InstrumentViewPresenter(ObservingPresenter):
     """
     Presenter holding the view widget for the InstrumentView.
     It has no model as its an old widget written in C++ with out MVP
     """
+
     view = None
 
-    def __init__(self, ws, parent=None):
-        self.view = InstrumentView(self, ws.name(), parent)
+    def __init__(self, ws, parent=None, ads_observer=None):
+        self.ws_name = ws.name()
```

Comment:
Do we not need to call `__init__`?

Reply:
The `ObservingPresenter`'s __init__ is empty

Reply:
Should the `InstrumentViewPresenter` necessarily know that? It seems like good practice to make sure the base class inits are called?